src/crafting.cpp

```diff
@@ -1293,6 +1307,20 @@ void pick_recipes(const inventory &crafting_inv,
                         continue;
                     }
                 }
+                if(search_qualities) {
+                  bool match_found = false;
+                  itype *it = item::find_type(rec->result);
+                  for(std::map<std::string, int>::const_iterator quality = it->qualities.begin(); quality != it->qualities.end() && !match_found; ++quality) {
+                    std::string qname = quality::get_name(quality->first);
+                    if (lcmatch(qname, filter)) {
+                      match_found = true;
+                    }
+                  }
+
+                  if (!match_found) {
```

Comment:
The check is somehow pointless, the loop would continue anyway (until it's stopped by the loop condition), it's like `if(...) continue; else continue;`.